src/map/utils/battleutils.cpp

```diff
@@ -4294,15 +4294,28 @@ namespace battleutils
         }
     }
 
-    void HandleTacticalParry(CBattleEntity* PEntity) {
-        for (uint8 j = 0; j < PEntity->TraitList.size(); ++j)
+    void HandleTacticalParry(CBattleEntity* PEntity)
+    {
+        for (const CTrait* PTrait : PEntity->TraitList)
```

Comment:
Find won't work here because it's std::vector<CTrait*>. But at least I added a break; :F

Reply:
If only it was a map..

Reply:
Don't you have that backwards? Find will work here *because* it's a vector.  If it were a map you wouldn't need find, because you'd just have the key.

Reply:
Yes and no. Find works if we’re looking for the data type that the vector stores. But we’re not. We need the trait based on ID so we need to manually loop through and look for it. Also, trait lists are not very large, not like mods, so looping through isn’t that bad.